test/test_torch.py

```diff
@@ -4151,12 +4151,9 @@ def run_test(shape, cast, diagonal):
                             self.assertEqual(x_nc.tril_(diagonal), exp_nc, 0)
 
                         # any 3-dimensional tensor should be fine
-                        if len(shape) <= 3 or s == -2:
+                        if len(shape) <= 3 or s == -2 or s < -3:
                             self.assertFalse(x_nc.is_contiguous(),
                                              "x_nc should remain non-contiguous")
-                        elif s < -3:
-                            self.assertTrue(x_nc.is_contiguous(),
-                                            "x_nc should become contiguous")
```

Comment:
Since making a non-contiguous tensor contiguous is always guaranteed to be an out-of-place operation, which breaks the in-place property of `triu_` and `tril_`, we should never expect `x_nc` to be made contiguous in `triu_` and `tril_`.